[FEATURE REQUEST] Show same remove dialog for every kind of file or folder

Aitorbp opened the pull request:
## Related Issues
App: https://github.com/owncloud/android/issues/4377

- [ ] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
- [ ] Add feature to Release Notes in `ReleaseNotesViewModel.kt` creating a new `ReleaseNote()` with String resources (if required)

_____

## QA

joragua commented on owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt:
```diff
@@ -1158,12 +1151,8 @@ class MainFileListFragment : Fragment(),
-                    filesToRemove = checkedFiles
-                    fileOperationsViewModel.showRemoveDialog(filesToRemove)
-                }
+                filesToRemove = checkedFiles
+                fileOperationsViewModel.showRemoveDialog(filesToRemove)
```
It can be used `checkedFiles` on `showRemoveDialog` instead of having a variable called `filesToRemove`

Aitorbp replied:
Ok, we could remove it.

joragua commented on owncloudApp/src/main/java/com/owncloud/android/presentation/files/removefile/RemoveFilesDialogFragment.kt:
```diff
@@ -90,44 +75,78 @@ class RemoveFilesDialogFragment : ConfirmationDialogFragment(), ConfirmationDial
+            messageStringId = if (targetFiles.size == 1) {
                 // choose message for a single file
-                val file = files.first()
+                val file = targetFiles.first()
+                messageArguments = file.fileName
```
We can delete  `file` variable and use `files.first().fileName` on the value of `messageArguments`

Aitorbp replied:
In this case I would leave it with my approach. If you notice, the `file` variable is later used to know if the file is a folder, therefore we cannot do without it.
Here the code:
```
val file = targetFiles.first()
                messageArguments = file.fileName
                if (file.isFolder) {
                    R.string.confirmation_remove_folder_alert
                } else {
                    R.string.confirmation_remove_file_alert
                }
```

joragua commented on owncloudApp/src/main/res/values/strings.xml:
```diff
@@ -760,6 +760,8 @@
     <string name="release_notes_4_3_0_title_show_app_provider_icon_from_endpoint">New icons in "Open in (web)" option on the operations menu</string>
     <string name="release_notes_4_3_0_subtitle_show_app_provider_icon_from_endpoint">More appropriate icons have been added to the "Open in (web)" option on the operations menu</string>
+    <string name="release_notes_4_3_0_title_show_same_remove_dialog_for_every_file_folder">Improved thumbnails in the delete files dialog</string>
+    <string name="release_notes_4_3_0_subtitle_show_same_remove_dialog_for_every_file_folder">Thumbnails have been added in remove dialog for all files and not only for images</string>
```
```suggestion
    <string name="release_notes_4_3_0_subtitle_show_same_remove_dialog_for_every_file_folder">Thumbnails have been added in remove dialog for all files and not only for items which had thumbnail</string>
```

Aitorbp replied:
Updated

joragua commented on changelog/unreleased/4404:
```diff
@@ -0,0 +1,7 @@
+Enhancement: Unified dialogs to remove files
+
+The logic of the custom remove files dialog has been incorporated into the RemoveFilesDialogFragment, unifying both developments.
+Thumbnails have been added for all files and not only for images.
```
```suggestion
The logic of the custom remove files dialog has been incorporated into the RemoveFilesDialogFragment, unifying both developments.
Thumbnails have been added for all files and not only for items which had thumbnail.
```

Aitorbp replied:
Updated

joragua reviewed (changes requested):
Some changes here

Aitorbp commented:
Add new shortcut icon to remove dialog when shortcut pr will be merged.

JuancaG05 commented on changelog/unreleased/4404:
```diff
@@ -0,0 +1,9 @@
+The custom dialog has been modified when the file that is going to be deleted has thumbnail.
+Also, when removing files in multiple selection, the number of elements that are going to be removed is displayed in the dialog.
+
+https://github.com/owncloud/android/issues/4377
+https://github.com/owncloud/android/issues/4342
+https://github.com/owncloud/android/pull/4348
+https://github.com/owncloud/android/pull/4404
```
Let's try to keep a bit of order here. If you added the new PR in second place, let's do the same with the issue, otherwise they won't be corresponding

Aitorbp replied:
Changed the order.

JuancaG05 commented on owncloudApp/src/main/java/com/owncloud/android/presentation/files/removefile/RemoveFilesDialogFragment.kt:
```diff
@@ -22,110 +22,127 @@
+    companion object {
+        const val TAG_REMOVE_FILES_DIALOG_FRAGMENT = "TAG_REMOVE_FILES_DIALOG_FRAGMENT"
+
+        /**
+         * Public factory method to create new RemoveFilesDialogFragment instances.
+         *
+         * @param files           Files to remove.
+         * @param isAvailableLocallyAndNotAvailableOffline
+         * @return Dialog ready to show.
+         */
```
Same as in other PRs: if you want to keep the docs, do it correctly. `isAvailableLocallyAndNotAvailableOffline` is not described

JuancaG05 commented on owncloudApp/src/main/java/com/owncloud/android/presentation/files/removefile/RemoveFilesDialogFragment.kt:
```diff
@@ -22,110 +22,127 @@
+         * @param isAvailableLocallyAndNotAvailableOffline
+         * @return Dialog ready to show.
+         */
+        @JvmStatic
```
Is this needed?

Aitorbp replied:
Removed

JuancaG05 commented on owncloudApp/src/main/res/values/strings.xml:
```diff
@@ -748,7 +748,7 @@
     <string name="release_notes_4_3_0_subtitle_clear_data_button_hard_reset">The "Clear data" button has been enabled to delete the application data from the app settings in the device settings. The application will perform a hard reset if this button is pressed</string>
     <string name="release_notes_4_3_0_title_improvements_remove_dialog">Improvements in remove dialog</string>
-    <string name="release_notes_4_3_0_subtitle_improvements_remove_dialog">Added a custom dialog when the file that is going to be deleted has thumbnail and added the number of files that are going to be deleted in a multiple selection</string>
+    <string name="release_notes_4_3_0_subtitle_improvements_remove_dialog">Modified the custom dialog when the file that is going to be deleted has thumbnail and added the number of files that are going to be deleted in a multiple selection</string>
```
Same as in calens...

Aitorbp replied:
changed

JuancaG05 commented on owncloudApp/src/main/res/values/strings.xml:
```diff
@@ -765,6 +765,8 @@
     <string name="release_notes_4_3_0_subtitle_show_app_provider_icon_from_endpoint">More appropriate icons have been added to the "Open in (web)" option on the operations menu</string>
     <string name="release_notes_4_3_0_title_av_offline_files_removed_locally_with_local_only_option">Displayed \"Local only\" option properly and prevented deletion of available offline files in the remove dialog</string>
     <string name="release_notes_4_3_0_subtitle_av_offline_files_removed_locally_with_local_only_option">\"Local only\" option in remove dialog will be shown when the selected folder contains at least one downloaded file, ignoring those available offline. If the "Local only" option is displayed and clicked, available offline files will not be deleted</string>
+    <string name="release_notes_4_3_0_title_show_same_remove_dialog_for_every_file_folder">Improved thumbnails in the delete files dialog</string>
+    <string name="release_notes_4_3_0_subtitle_show_same_remove_dialog_for_every_file_folder">Thumbnails have been added in remove dialog for all files and not only for images and .txt files</string>
```
You shouldn't add a new release note, just modifying the one that was already there regarding the remove dialog

Aitorbp replied:
deleted release note repeated

JuancaG05 reviewed (changes requested):
Several things to check here @Aitorbp

Aitorbp commented on changelog/unreleased/4404:
```diff
@@ -0,0 +1,9 @@
+Enhancement: Improvements in remove dialog
+
+The custom dialog has been modified when the file that is going to be deleted has thumbnail.
```
Deleted

JuancaG05 commented on changelog/unreleased/4404:
```diff
@@ -1,7 +1,9 @@
+Enhancement: Improvements in remove dialog
 
-A custom dialog alert has been added when the file that is going to be deleted has thumbnail.
+The dialog has been modified when the file that is going to be deleted has thumbnail.
```
We don't know which dialog this is referring to, you can't write "The dialog...". Also, now we use this dialog no matter whether the file has thumbnail or not. Please, take a look at this carefully

JuancaG05 commented on changelog/unreleased/4404:
```diff
@@ -1,7 +1,9 @@
 Also, when removing files in multiple selection, the number of elements that are going to be removed is displayed in the dialog.
 
-https://github.com/owncloud/android/issues/4342
+https://github.com/owncloud/android/issues/4377
 https://github.com/owncloud/android/pull/4348
+https://github.com/owncloud/android/issues/4342
+https://github.com/owncloud/android/pull/4404
```
What sense does this order make?

Aitorbp replied:
Modified

JuancaG05 commented on owncloudApp/src/main/res/values/strings.xml:
```diff
@@ -748,7 +748,7 @@
     <string name="release_notes_4_3_0_subtitle_clear_data_button_hard_reset">The "Clear data" button has been enabled to delete the application data from the app settings in the device settings. The application will perform a hard reset if this button is pressed</string>
     <string name="release_notes_4_3_0_title_improvements_remove_dialog">Improvements in remove dialog</string>
-    <string name="release_notes_4_3_0_subtitle_improvements_remove_dialog">Added a custom dialog when the file that is going to be deleted has thumbnail and added the number of files that are going to be deleted in a multiple selection</string>
+    <string name="release_notes_4_3_0_subtitle_improvements_remove_dialog">Modified the dialog when the file that is going to be deleted has thumbnail and added the number of files that are going to be deleted in a multiple selection</string>
```
Same as in calens... Why modified? Users won't have seen the previous release note by the time they see this one, so we can not say that it has been modified because for them this is going to be brand new! Also, wasn't this PR about showing the dialog for EVERY file, not only for those that have thumbnail as the release note says? We're not giving correct information here

Aitorbp replied:
Changed

JuancaG05 reviewed (changes requested):
Please, again, remove that new release note which shouldn't be here...

jesmrec commented:
QA checks:

- [ ] Remove folder
- [ ] Remove txt
- [ ] Remove pic
- [ ] Remove video
- [ ] Remove audio
- [ ] Remove pdf
- [ ] Remove office (docx, pptx, xlsx)
- [ ] Remove shortcut
- [ ] Remove zip
- [ ] Remove html
- [ ] Remove docxf
- [ ] Remove unknown
- [ ] Remove multiselection

jesmrec commented:
### (1)

1. Select any item on the file list and click on `Remove` 
2. Change device orientation

**Current:**

app crashes. This is the stacktrace:

```
Process: com.owncloud.android.debug, PID: 29844
                 java.lang.RuntimeException: Unable to start activity ComponentInfo{com.owncloud.android.debug/com.owncloud.android.ui.activity.FileDisplayActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.owncloud.android.presentation.files.removefile.RemoveFilesDialogFragment: could not find Fragment constructor
                 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
                 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
                 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5456)
                 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5362)
                 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
                 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
                 	at android.os.Handler.dispatchMessage(Handler.java:106)
                 	at android.os.Looper.loop(Looper.java:223)
                 	at android.app.ActivityThread.main(ActivityThread.java:7656)
                 	at java.lang.reflect.Method.invoke(Native Method)
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                 Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.owncloud.android.presentation.files.removefile.RemoveFilesDialogFragment: could not find Fragment constructor
                 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:681)
                 	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
                 	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:507)
                 	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:81)
                 	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
                 	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2507)
                 	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:2667)
                 	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117)
                 	at androidx.fragment.app.FragmentActivity.lambda$init$3$androidx-fragment-app-FragmentActivity(FragmentActivity.java:140)
                 	at androidx.fragment.app.FragmentActivity$$ExternalSyntheticLambda3.onContextAvailable(Unknown Source:2)
                 	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
                 	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:352)
                 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:217)
                 	at com.owncloud.android.ui.activity.DrawerActivity.onCreate(DrawerActivity.kt:588)
                 	at com.owncloud.android.ui.activity.FileActivity.onCreate(FileActivity.java:119)
                 	at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(FileDisplayActivity.kt:197)
                 	at android.app.Activity.performCreate(Activity.java:8000)
                 	at android.app.Activity.performCreate(Activity.java:7984)
                 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
                 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
                 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
                 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5456) 
                 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5362) 
                 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69) 
                 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
                 	at android.os.Handler.dispatchMessage(Handler.java:106) 
                 	at android.os.Looper.loop(Looper.java:223) 
                 	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                 	at java.lang.reflect.Method.invoke(Native Method) 
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
                 Caused by: java.lang.NoSuchMethodException: com.owncloud.android.presentation.files.removefile.RemoveFilesDialogFragment.<init> []
                 	at java.lang.Class.getConstructor0(Class.java:2332)
                 	at java.lang.Class.getConstructor(Class.java:1728)
                 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:666)
```

**Expected:** 

no crash


Pixel 2, Android 11
`015c716f1`

Aitorbp commented:
The last report is ready to test @jesmrec. Also I added in another commit one fix related to release note of shortcut. I added here because that [pr](https://github.com/owncloud/android/pull/4420) is already merged.

jesmrec commented:
Crash is fixed and feature is OK from my side.

About this 

> Also I added in another commit one fix related to release note of shortcut. I added here because that pr is already merged.

I'd change to avoid the word `FAB` in the release notes, that's pretty technically scoped to Android and non-advanced users will not understand.

Aitorbp commented:
> Crash is fixed and feature is OK from my side.
> 
> About this
> 
> > Also I added in another commit one fix related to release note of shortcut. I added here because that pr is already merged.
> 
> I'd change to avoid the word `FAB` in the release notes, that's pretty technically scoped to Android and non-advanced users will not understand.

Changed by `floating button`

jesmrec commented:
Approved
